### PR TITLE
icestorm: Initial inclusion in the repository

### PR DESCRIPTION
**Pull request by infinitymdm**
**Summary**

Initial inclusion of icestorm bitstream tools and nextpnr place-and-route tools for Lattice iCE40 FPGAs.
Inclusion of icestorm resolves https://github.com/getsolus/packages/issues/142
Inclusion of nextpnr resolves https://github.com/getsolus/packages/issues/143

**Test Plan**

Synthesized a simple SystemVerilog counter project and tested it on an Alchitry Cu FPGA. 

**Checklist**

- [x] Package was built and tested against unstable

### CODE REVIEW

**Comment by infinitymdm** [comment]
Rebuilt against current libboost

File: packages/i/icestorm/pspec_x86_64.xml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        <Archive type="binary" sha1sum="79eb0752a961b8e0d15c77d298c97498fbc89c5a">https://sources.getsol.us/README.Solus</Archive>
+    </Source>
+    <Package>
+        <Name>icestorm</Name>
+        <Summary xml:lang="en">A collection of tools and documentation for iCE40 FPGAs.</Summary>
+        <Description xml:lang="en">Project IceStorm aims at documenting the bitstream format of Lattice iCE40 FPGAs and providing simple tools for analyzing and creating bitstream files. See https://clifford.at/icestorm for more information.
+</Description>
+        <PartOf>office.scientific</PartOf>
+        <Files>
+            <Path fileType="config">/etc/udev/rules.d/53-lattice-ftdi.rules</Path>
```
Comment (ReillyBrogan) [inline packages/i/icestorm/pspec_x86_64.xml:23]: Use `/usr/lib/udev/rules.d/` for udev rules shipped by packages.

File: packages/n/nextpnr/package.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    - generic : /usr/bin/*-generic
+builddeps  :
+    - pkgconfig(Qt5Core)
+    - pkgconfig(eigen3)
+    - pkgconfig(python3)
+    - icestorm
+    - libboost-devel
+    - yosys
+setup      : |
+    %cmake \
```
Comment (ReillyBrogan) [inline packages/n/nextpnr/package.yml:23]: Use `%cmake_ninja` unless for whatever reason this doesn't work with the ninja build generator.

File: packages/n/nextpnr/package.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
+name       : nextpnr
```
Comment (ReillyBrogan) [inline packages/n/nextpnr/package.yml:1]: You have two package updates in one commit, please split these to separate commits.

File: packages/i/icestorm/package.yml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+release    : 1
+source     :
+    - git|https://github.com/YosysHQ/icestorm : d20a5e9001f46262bf0cef220f1a6943946e421d
+homepage   : https://clifford.at/icestorm
+license    : ISC
+component  : office.scientific
+summary    : A collection of tools and documentation for iCE40 FPGAs.
+description: |
+    Project IceStorm aims at documenting the bitstream format of Lattice iCE40 FPGAs and providing simple tools for analyzing and creating bitstream files. See https://clifford.at/icestorm for more information.
+strip      : no
```
Comment (ReillyBrogan) [inline packages/i/icestorm/package.yml:12]: Are you _sure_ you need `strip: no`?

File: packages/n/nextpnr/package.yml [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+description: |
+    nextpnr is a vendor-neutral, timing-driven, FOSS FPGA placement and routing tool.
+patterns   :
+    - ice40 : /usr/bin/*-ice40
+    - generic : /usr/bin/*-generic
+builddeps  :
+    - pkgconfig(Qt5Core)
+    - pkgconfig(eigen3)
+    - pkgconfig(python3)
+    - icestorm
```
Comment (ReillyBrogan) [inline packages/n/nextpnr/package.yml:19]: You're using icestorm as a builddep only, is that correct? You don't need it installed at runtime with the `nextpnr` package?

**Review by ReillyBrogan — CHANGES_REQUESTED** [review]

File: packages/n/nextpnr/package.yml (hunk 3 above)
Comment (infinitymdm) [reply packages/n/nextpnr/package.yml:1]: What's the best way to do this? Do a hard reset and then add two separate commits?

Comment (ReillyBrogan) [reply packages/n/nextpnr/package.yml:1]: https://stackoverflow.com/questions/6217156/break-a-previous-commit-into-multiple-commits

File: packages/n/nextpnr/package.yml (hunk 5 above)
Comment (infinitymdm) [reply packages/n/nextpnr/package.yml:19]: icestorm isn't strictly required for nextpnr-ice40 to work, but most flows will use both tools. There is an argument to be made for including it as a rundep, as that's probably what people will expect. I'll add it in.

**Review by ReillyBrogan — APPROVED** [review]
